slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/JdbcMapperTest.scala

```diff
@@ -76,6 +76,50 @@ class JdbcMapperTest extends AsyncTest[JdbcTestDB] {
     )
   }
 
+  def testHeadOption = {
+    import slick.util._
+
+    case class Data(a: Int, b: Int, c: Option[String])
+
+    class T(tag: Tag) extends Table[Data](tag, "T_headOption") {
+      def a = column[Int]("A")
+      def b = column[Int]("B")
+      def c = column[Option[String]]("C")
+      def * = (a, b, c).mapTo[Data]
+    }
+    val ts = TableQuery[T]
+
+    val headQ1 = ts.filter(_.a === 10).headOption
+    val headQ2 = ts.filter(_.b === 3).map(_.a + 1).headOption
+    val headQ3 = ts.filter(_.a === 2).map(_.b).headOption
+    val headQ4 = ts.filter(_.a === 1).map(_.c).headOption
```

Comment:
It would be useful to actually test these types:

    val headQ4 = ts.filter(_.a === 1).map(_.c).headOption
    val headQ4t: Rep[Option[Option[String]]] = headQ4

 There should be a test for record types as well, e.g.:

    val headQ5 = ts.filter(_.a === 1).map(t => (t.a, t.c)).headOption
    val headQ5t: Rep[Option[(Rep[Int], Rep[Option[String]])]] = headQ5